js/utilities.js

```diff
@@ -894,31 +894,49 @@
      *                            Or (The entry cannot be created. Please use a combination of different _fields_ to create new record.)
      *
      */
-    module._conflictErrorMapping = function(errorStatusText, generatedErrMessage) {
-      var mappedErrMessage;
+    module._conflictErrorMapping = function(errorStatusText, generatedErrMessage, reference, actionFlag) {
+      var mappedErrMessage, refTable, tableDisplayName = '';
+      var ref = reference;
       var conflictErrorPrefix = "409 Conflict\nThe request conflicts with the state of the server. ";
 
-      if (generatedErrMessage.indexOf("violates foreign key constraint") > -1) {
+      if (generatedErrMessage.indexOf("violates foreign key constraint") > -1 && actionFlag == "DEL") {
 
           var referenceTable = "another";
-          
+
           var detail = generatedErrMessage.search(/DETAIL:/g);
           if (detail > -1) {
             detail = generatedErrMessage.substring(detail, generatedErrMessage.length);
             referenceTable = detail.match(/referenced from table \"(.*)\"(.*)/);
             if(referenceTable && referenceTable.length > 1){
-                referenceTable =  "the <code>"+ referenceTable[1] +"</code>";
+                refTable = referenceTable[1];
+                referenceTable =  refTable;
             }
           }
 
+          // if(){
+            var fkConstraint = generatedErrMessage.match(/foreign key constraint \"(.*?)\"/)[1];    //get constraintName
+            if(fkConstraint != 'undefined' && fkConstraint != ''){
+              var relatedRef = ref.related(); //get all related references
+
+              for(i = 0; i < relatedRef.length; i++){
+                  key  = relatedRef[i];
+                  if(key.origFKR.constraint_names["0"][1] == fkConstraint && key.origFKR._table.name == refTable){
```

Comment:
change this from `"0"` to `0`

Reply:
Add a flag that you shows whether you found the referenceTable or not. If not. append the "If you have trouble removing dependencies please contact the site administrator." message.

**Please read all the comments in the issue before creating PR.**

Reply:
Added site Admin message by default and if the table name is found in `displayname`then it will be replaced by blank string in the main message.